components/brave_wallet/brave_wallet_utils.cc

```diff
@@ -133,4 +137,48 @@ std::string Uint256ValueToHex(uint256_t input) {
   return out;
 }
 
+std::string GenerateMnemonic(const std::vector<uint8_t>& entropy) {
+  uint8_t* entropy_in = const_cast<uint8_t*>(entropy.data());
+  size_t size_in = entropy.size();
+
+  // If input entropy is empty, we generate a random 128 bits
+  std::vector<uint8_t> bytes(16);
+  if (entropy.empty()) {
+    crypto::RandBytes(&bytes[0], bytes.size());
+    entropy_in = bytes.data();
+    size_in = bytes.size();
+  }
+
+  // entropy size should be 128-256 bits
+  if (size_in < 16 || size_in > 32) {
+    LOG(ERROR) << __func__ << ": Entropy should be 128-256 bits";
+    return "";
+  }
+
+  char* words = nullptr;
+  std::string result;
+  if (bip39_mnemonic_from_bytes(nullptr, entropy_in, size_in, &words) !=
+      WALLY_OK) {
+    LOG(ERROR) << __func__ << ": bip39_mnemonic_from_bytes failed";
+    return result;
+  }
+  result = words;
+  wally_free_string(words);
+
+  return result;
+}
+
+std::unique_ptr<std::vector<uint8_t>> MnemonicToSeed(
+    const std::string& mnemonic,
+    const std::string& passphrase) {
```

Comment:
Should we sanitize the passphrase to be of the form `UTF-8 NFKD`? I am uncertain myself if this is needed. The spec says the passphrase must be of this form.

Reply:
I think we don't need this for now since we pass the test vector and then in the use of our HD wallet. When generating seed, the passphrase will be empty and the default salt is ACSII, matching the behavior of Metamask